Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3-alpine AS base
+
+WORKDIR /opt/app
+
+# Install Poetry
+RUN pip install pipx
+RUN pipx install poetry
+RUN ln -s ~/.local/bin/poetry /usr/bin/poetry
+
+# Install Dependencies
+COPY ./poetry.lock ./pyproject.toml ./poetry.toml ./
+RUN poetry install
+
+# Copy App
+COPY ./todo_app/ ./todo_app/
+
+
+FROM base AS prod
+
+ENV FLASK_DEBUG=false
+
+CMD [ "poetry", "run", "gunicorn", "--bind=0.0.0.0:5000", "todo_app.app:create_app()" ]
```

Comment:
Any reason you've defaulted to CMD rather than ENTRYPOINT here? [The table agrees](https://docs.docker.com/reference/dockerfile/#understand-how-cmd-and-entrypoint-interact) they're functionally identical but my own preference is always for ENTRYPOINT because it matches the intent, e.g. if we read those docs it says:
> ENTRYPOINT should be defined when using the container as an executable.

Reply:
I'm not sure this application counts as an "executable", is my reason for using `CMD`.

It means I'm able to run `docker run <image> /bin/sh` and get a shell without having to specify `--entrypoint`

I would almost definitely use `ENTRYPOINT` for something like `jq` as adding some arguments after the `docker run` I feel should pass into `jq` itself. As this isn't an executable that takes arguments (or that's not its main thing), `CMD` feels nicer here.

Reply:
https://stackoverflow.com/a/34245657

I suppose I've always just found `docker run <image> /bin/bash` very useful